tensor/stats/cluster/cluster.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2024, Cogent Core. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package cluster
+
+//go:generate core generate
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+
+	"cogentcore.org/core/base/indent"
+	"cogentcore.org/core/tensor"
+	"cogentcore.org/core/tensor/stats/stats"
+)
+
+// todo: all of this data goes into the datafs
+// Cluster makes a new dir, stuffs results in there!
+// need a global "cwd" that it uses, so basically you cd
+// to a dir, then cal it.
+
+// Node is one node in the cluster
+type Node struct {
+	// index into original distance matrix; only valid for for terminal leaves.
+	Index int
+
+	// Distance value for this node, i.e., how far apart were all the kids from
+	// each other when this node was created. is 0 for leaf nodes
+	Dist float64
+
+	// ParDist is total aggregate distance from parents; The X axis offset at which our cluster starts.
+	ParDist float64
+
+	// Y is y-axis value for this node; if a parent, it is the average of its kids Y's,
+	// otherwise it counts down.
+	Y float64
+
+	// Kids are child nodes under this one.
+	Kids []*Node
+}
+
+// IsLeaf returns true if node is a leaf of the tree with no kids
+func (nn *Node) IsLeaf() bool {
+	return len(nn.Kids) == 0
+}
+
+// Sprint prints to string
+func (nn *Node) Sprint(labels *tensor.Indexed, depth int) string {
```

Comment:
`String`?

Reply:
not stringer, and not relevant b/c datafs.